engine/lib/output.php

```diff
@@ -142,6 +142,11 @@ function elgg_format_attributes(array $attrs = array()) {
 	$attributes = [];
 
 	foreach ($attrs as $attr => $val) {
+		if (!is_string($attr) || $attr[0] == '#') {
```

Comment:
If we're going to push arbitrary keys into input views (I get the use case), we should consolidate them as `#field_vars` or something and document them in the input views.